e2e/main_test.go

```diff
@@ -107,33 +105,35 @@ func mockEnvoyVersionsServer() (*httptest.Server, error) {
 	return ts, nil
 }
 
-// readFuncEPath reads E2E_FUNC_E_BINARY or defaults to "$PWD/dist/func-e_$GOOS_$GOARCH/func-e"
+// readFuncEBin reads E2E_FUNC_E_PATH or defaults to "../func-e"
 // An error is returned if the value isn't an executable file.
-func readFuncEPath() (string, error) {
-	path := os.Getenv(funcEBinaryEnvKey)
-	if path == "" {
-		// Assemble the default created by "make bin"
-		relativePath := filepath.Join("..", "dist", fmt.Sprintf("func-e_%s_%s", runtime.GOOS, runtime.GOARCH), "func-e")
-		abs, err := filepath.Abs(relativePath)
-		if err != nil {
-			return "", fmt.Errorf("%s didn't resolve to a valid path. Correct environment variable %s", path, funcEBinaryEnvKey)
+func readFuncEBin() error {
+	path := os.Getenv(funcEPathEnvKey)
+	path = filepath.Join("..", path) // relative to project root
+	if path != "" {
+		stat, err := os.Stat(path)
+		if err != nil && os.IsNotExist(err) {
+			return fmt.Errorf("%s doesn't exist. Correct environment variable %s", path, funcEPathEnvKey)
+		}
+		if !stat.IsDir() {
+			return fmt.Errorf("%s is not a directory. Correct environment variable %s", path, funcEPathEnvKey)
 		}
-		path = abs
 	}
 
-	stat, err := os.Stat(path)
-	if err != nil && os.IsNotExist(err) {
-		return "", fmt.Errorf("%s doesn't exist. Correct environment variable %s", path, funcEBinaryEnvKey)
-	}
-	if stat.IsDir() {
-		return "", fmt.Errorf("%s is not a file. Correct environment variable %s", path, funcEBinaryEnvKey)
+	abs, err := filepath.Abs(path)
+	if err != nil {
+		return fmt.Errorf("%s didn't resolve to a valid path. Correct environment variable %s", path, funcEPathEnvKey)
 	}
+	funcEBin = filepath.Join(abs, "func-e"+moreos.Exe)
+	stat, _ := os.Stat(funcEBin)
+
 	// While "make bin" should result in correct permissions, double-check as some tools lose them, such as
 	// https://github.com/actions/upload-artifact#maintaining-file-permissions-and-case-sensitive-files
-	if !moreos.IsExecutable(stat) {
-		return "", fmt.Errorf("%s is not executable. Correct environment variable %s", path, funcEBinaryEnvKey)
+	if stat == nil || !moreos.IsExecutable(stat) {
+		return fmt.Errorf("%s is not executable. Run `go build .` or `make bin` %s", funcEBin, funcEPathEnvKey)
```

Comment:
nit: if stat == nil, then `%s is not executable` sounds a bit confusing to me since that case os.Stat itself fails.

Reply:
well it is also not executable in that case, was trying to save 3 more lines of copy/pasta

Reply:
fair enough!

Reply:
fixing it